### PR TITLE
Add options to enable vboot

### PR DESCRIPTION
**Pull request by IgorBagnucki**
Signed-off-by: Igor Bagnucki <igor.bagnucki@3mdeb.com>

### CODE REVIEW

File: src/mainboard/protectli/vault_cml/Kconfig [hunk 1]
```diff
@@ -46,6 +47,22 @@ config VGA_BIOS_ID
...
+config VBOOT
+	select GBB_FLAG_DISABLE_EC_SOFTWARE_SYNC
+	select GBB_FLAG_DISABLE_FWMP
+	select GBB_FLAG_DISABLE_LID_SHUTDOWN
+	select GBB_FLAG_DISABLE_PD_SOFTWARE_SYNC
+	select VBOOT_ALWAYS_ENABLE_DISPLAY
+	select VBOOT_NO_BOARD_SUPPORT
+
+config VBOOT_SLOTS_RW_A
+	default y
```
Comment (miczyg1) [inline src/mainboard/protectli/vault_cml/Kconfig:59]:
```suggestion
	default y if VBOOT
```

File: src/mainboard/protectli/vault_cml/vboot-rwa.fmd [hunk 2]
```diff
@@ -0,0 +1,38 @@
+FLASH@0xff000000 0x1000000 {
```
Comment (miczyg1) [inline src/mainboard/protectli/vault_cml/vboot-rwa.fmd:1]: The best example of fmd file is here: https://github.com/Dasharo/coreboot/blob/asus_kgpe-d16/release/src/mainboard/asus/kgpe-d16/vboot-rwa-16M.fmd

You should specify regions with fixed size and provide the exact sizes. The CBFS regions should not have any sized defined to let flashmap parser spread it across whole free space. Of course you need to maintain DESC nad ME region. WP_RO must be aligned with the protected regions of SPI flash. Check the datasheet of the flash chip and verify the WP_RO length

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: src/mainboard/protectli/vault_cml/Kconfig [hunk 3]
```diff
@@ -3,16 +3,17 @@ if BOARD_PROTECTLI_VP4620
...
 	select BOARD_ROMSIZE_KB_16384
-	select SOC_INTEL_COMETLAKE_1
 	select HAVE_ACPI_RESUME
 	select HAVE_ACPI_TABLES
-	select SOC_INTEL_COMMON_BLOCK_HDA_VERB
-	select SOC_INTEL_COMMON_BLOCK_HDA
-	select MAINBOARD_HAS_LPC_TPM
-	select SUPERIO_ITE_IT8786E
 	select INTEL_GMA_HAVE_VBT
+	select MAINBOARD_HAS_LPC_TPM
```
[the remark this answers is not in the document]
Comment (IgorBagnucki) [reply src/mainboard/protectli/vault_cml/Kconfig]: Fixed this. Now it should be in the original order.

File: src/mainboard/protectli/vault_cml/Kconfig (hunk 1 above)
Comment (IgorBagnucki) [reply src/mainboard/protectli/vault_cml/Kconfig]: Fixed

File: src/mainboard/protectli/vault_cml/vboot-rwa.fmd (hunk 2 above)
Comment (IgorBagnucki) [reply src/mainboard/protectli/vault_cml/vboot-rwa.fmd:1]: I have aligned the RO section with SPI flash acording to the [datasheet](https://datasheetspdf.com/pdf-file/850581/Macronix/25L12835F/1)

Comment (miczyg1) [reply src/mainboard/protectli/vault_cml/vboot-rwa.fmd:1]: Now looks good.

**Comment by miczyg1** [comment]
Why still draft?

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**